cms/src/main/java/uk/nhs/digital/ps/chart/HighchartsConfigDerivedDataFunction.java

```diff
@@ -0,0 +1,135 @@
+package uk.nhs.digital.ps.chart;
+
+import static java.util.Collections.emptyMap;
+import static org.apache.commons.lang3.ObjectUtils.allNotNull;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.hippoecm.repository.HippoStdNodeType;
+import org.hippoecm.repository.ext.DerivedDataFunction;
+import org.onehippo.cms7.services.HippoServiceRegistry;
+import org.slf4j.Logger;
+
+import java.util.Map;
+import javax.jcr.Binary;
+import javax.jcr.Value;
+
+public class HighchartsConfigDerivedDataFunction extends DerivedDataFunction {
+
+    private static final long serialVersionUID = 1;
+
+    private static final Logger log = getLogger(HighchartsConfigDerivedDataFunction.class);
+
+    private static final String PARAM_KEY_VARIANT = "variant";
+    private static final String PARAM_KEY_DATA = "data";
+    private static final String PARAM_KEY_TYPE = "type";
+    private static final String PARAM_KEY_TITLE = "title";
+    private static final String PARAM_KEY_Y_TITLE = "yTitle";
+    private static final String PARAM_KEY_CHART_CONFIG = "chartConfig";
+
+    private final HighchartsInputParser parser;
+    private final ObjectMapper objectMapper;
+
+    public HighchartsConfigDerivedDataFunction() {
+        this(HippoServiceRegistry.getService(HighchartsInputParser.class), createObjectMapper());
```

Comment:
As discussed, how about we delegate the construction of dependencies to a factory class and then we can get rid of the Service

Reply:
As subsequently discussed, the module and the parser service registration were left in place after all, seeing that they were correctly leveraging the purpose of `HippoServiceRegistry` which is to provide common dependencies to other classes (to validator and to workflow task in this case).

Having said that, the derived function has been replaced with a workflow task and, thanks to the fact that the task is now being created by the workflow action, it was possible to eliminate the constructor previously created just to allow injection of mocks during tests.